File: Blog.Service/Controllers/PostController.cs
using Blog.Service.Base;
using Common.Enums;
using DataManager;
using DataTransfer;
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;

namespace Blog.Service.Controllers
{
    public class PostController : BaseController
    {
        private readonly PostManager _postManager;

        public PostController(PostManager postManager)
        {
            _postManager = postManager;
        }

        [HttpGet]
        public IEnumerable<PostDto> GetAll(Language language)
        {
            return _postManager.GetAllPosts(language);
        }

        [HttpPost]
        public IActionResult AddPost(AddPostDto dto)
        {
            return Ok(dto);
        }

        [HttpPost("Content")]
        public IActionResult AddPostContent(AddPostContentDto dto)
        {
            return Ok(dto);
        }
    }
}

File: Blog.Service/Controllers/PostController.fs
namespace Blog.Service.Controllers

open Microsoft.AspNetCore.Mvc
open Blog.Service.Services
open Blog.Service.Models
open System

[<ApiController>]
[<Route("api/[controller]")>]
type PostController (_postService: PostService) =
    inherit ControllerBase()

    [<HttpGet>]
    member this.Get() =
        this.Ok(_postService.Get())

    [<HttpPost>]
    member this.Create(post: Post) =
        this.Ok(_postService.Create(post)) :> IActionResult

    [<HttpPutAttribute>]
    member this.Update(post: Post) =
        raise (NotImplementedException())